### PR TITLE
Add tooltip to the explore card types

### LINKED ISSUE #863: [Feature Request]: Add Tooltip to Explore Card Types
### Feature description

Our Explore cards have a type of managed state, we should add a tooltip so users don't need to seek out documentation. This the content comes from the OdhApplication CRs (`.spec.category`).

- **Partner managed:** Partner managed software is hosted on the ISV’s cloud service.
- **Red Hat managed:** Red Hat managed software is hosted on Red Hat’s OSD cluster. 
- **Self-managed:** Self-managed software is installed to a RHODS cluster, but does not support upgrade testing, alerting, or other features of externally managed software.

### Describe alternatives you've considered

_No response_

### Anything else?

We have only a desire to have 3 types to this field, we may want to lock down the CRD to an enum (I think that's possible in the schema). 

Either way, if the value is not there or is not one of the 3, we should not show any tooltip.

We shouldn't be stringent to case sensitivity, so make sure the check is safe from that side too.

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
## Description

Closes #863 
Add a tooltip as described in the issue, and updated the CRD as well. At the same time, updated part of the code to make the badge show the correct color that they should be.

Screen Shot 2023-01-05 at 2 15 36 PM

## How Has This Been Tested?

Go to the explore page and hover over the type badges.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by lucferbux — APPROVED** [review]

File: frontend/src/components/OdhExploreCardTypeBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  let content;
+
+  if (category === OdhApplicationCategory.RedHatManaged) {
+    content = 'Red Hat managed software is hosted on Red Hat’s OSD cluster';
+  } else if (category === OdhApplicationCategory.PartnerManaged) {
+    content = 'Partner managed software is hosted on the ISV’s cloud service';
+  } else if (category === OdhApplicationCategory.SelfManaged) {
+    content =
+      'Self-managed software is installed to a RHODS cluster, but does not support upgrade testing, alerting, or other features of externally managed software';
+  } else {
```
Comment (andrewballantyne) [inline frontend/src/components/OdhExploreCardTypeBadge.tsx:19]: Small nit -- I get what you're doing here works... but I'd recommend actually doing this:

```
if - case A; set tooltip to 'A'
else if - case B; set tooltip to 'B'
else if - case C; set tooltip to 'C'

if !tooltip; return
```
That way if we add another case that doesn't set the tooltip for whatever reason we don't try to render a Tooltip with content.

It's the same thing right now, but it separates the category statements from the return -- so we can manage the fall-through case a bit less "in with the rest"

File: frontend/src/components/OdhAppCard.tsx [hunk 2]
```diff
@@ -92,7 +92,7 @@ const OdhAppCard: React.FC<OdhAppCardProps> = ({ odhApp }) => {
   ];
 
   const badgeClasses = classNames('odh-card__partner-badge', {
-    'm-warning': odhApp.spec.category === 'Third party support',
+    'm-warning': odhApp.spec.support === 'third party support',
```
Comment (andrewballantyne) [inline frontend/src/components/OdhAppCard.tsx:95]: I'm concerned about these support vs category changes -- also any reason they are not an enum?

File: frontend/src/components/OdhAppCard.tsx [hunk 3]
```diff
@@ -204,7 +204,7 @@ const OdhAppCard: React.FC<OdhAppCardProps> = ({ odhApp }) => {
       <CardBody>
         {!dashboardConfig.spec.dashboardConfig.disableISVBadges &&
         odhApp.spec.category &&
-        odhApp.spec.category !== ODH_PRODUCT_NAME ? (
+        odhApp.spec.support !== ODH_PRODUCT_NAME ? (
```
Comment (andrewballantyne) [inline frontend/src/components/OdhAppCard.tsx:207]: Would not support be "red hat" or "third party support" -- not RHODS or ODH?

File: frontend/src/components/OdhExploreCard.tsx [hunk 4]
```diff
@@ -45,8 +46,8 @@ const OdhExploreCard: React.FC<OdhExploreCardProps> = ({
   const disabled = odhApp.spec.comingSoon || disableInfo;
   const cardClasses = classNames('odh-card', { 'm-disabled': disabled });
   const badgeClasses = classNames('odh-card__partner-badge', {
-    'm-warning': odhApp.spec.category === 'Third party support',
-    'm-hidden': odhApp.spec.category === ODH_PRODUCT_NAME,
+    'm-warning': odhApp.spec.support === 'third party support',
+    'm-hidden': odhApp.spec.support === ODH_PRODUCT_NAME,
```
Comment (andrewballantyne) [inline frontend/src/components/OdhExploreCard.tsx:50]: Same question here...

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: manifests/crd/odhapplications.dashboard.opendatahub.io.crd.yaml [hunk 5]
```diff
@@ -43,6 +43,10 @@ spec:
                   type: string
                 category:
                   type: string
+                  enum:
```
Comment (andrewballantyne) [inline manifests/crd/odhapplications.dashboard.opendatahub.io.crd.yaml:46]: Undo this based on the conversation in slack from a few days ago. I'd like to see this make this sprint and they are hesitant about CRD changes due to CRs in flight out there

File: frontend/src/types.ts [hunk 6]
```diff
@@ -146,7 +146,7 @@ export type OdhApplication = {
     docsLink: string;
     getStartedLink: string;
     getStartedMarkDown: string;
-    category?: string;
+    category?: OdhApplicationCategory;
```
Comment (andrewballantyne) [inline frontend/src/types.ts:149]:
```suggestion
    category?: OdhApplicationCategory | string; // unbound by the CRD today -- should be the enum
```

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
/unhold

File: frontend/src/components/OdhExploreCardTypeBadge.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { Tooltip } from '@patternfly/react-core';
+import { OdhApplicationCategory } from '../types';
+
+type OdhExploreCardTypeBadgeProps = {
+  category: OdhApplicationCategory;
```
Comment (andrewballantyne) [inline frontend/src/components/OdhExploreCardTypeBadge.tsx]: You'll need this to also handle a string -- since it's unbounded.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Review by andrewballantyne — APPROVED** [review]
/hold

Works great, thanks for doing this! Sorry about the CRD back and forth -- didn't know they were so immutable.

If you want to fix the if-statement comment, please do -- I'll reapply the lgtm. If not, just unhold it. It's not critical in anyway.

**Review by andrewballantyne — APPROVED** [review]

**Comment by DaoDaoNoCode** [comment]
found one issue, hold for a moment.

**Comment by DaoDaoNoCode** [comment]
@andrewballantyne Can you review it again? I made a small mistake just now but now it's fixed.

**Comment by andrewballantyne** [comment]
> @andrewballantyne Can you review it again? I made a small mistake just now but now it's fixed.

Good catch -- I missed that.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**